lib/apiservers/engine/backends/volume.go

```diff
@@ -58,7 +105,43 @@ func (v *Volume) VolumeRm(name string) error {
 		if _, ok := err.(*storage.RemoveVolumeConflict); ok {
 			return derr.NewRequestConflictError(fmt.Errorf("Volume is in use"))
 		}
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error form portlayer"), http.StatusInternalServerError)
+		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error from portlayer: %s", err), http.StatusInternalServerError)
 	}
 	return nil
 }
+
+//Utility Functions
+
+func fillDockerVolumeModel(volume *models.VolumeResponse, labels map[string]string) *types.Volume {
+	dockerVol := types.Volume{
+		Driver:     volume.Driver,
+		Name:       volume.Name,
+		Labels:     labels,
+		Mountpoint: volume.SelfLink,
+	}
+	return &dockerVol
+}
+
+func translateInputsToPortlayerRequestModel(name, host, driverName string, opts, labels map[string]string) models.VolumeRequest {
+	model := models.VolumeRequest{
+		Driver:     driverName,
+		DriverArgs: opts,
+		Name:       name,
+		Store:      host,
+	}
+
+	metadata := createVolumeMetadata(&model, labels)
+	model.Metadata = make(map[string]string)
+	model.Metadata["dockerMetaData"] = metadata
+	return model
+}
+
+func createVolumeMetadata(model *models.VolumeRequest, labels map[string]string) string {
+	metadata := volumeMetadata{
+		Driver: model.Driver,
+		Name:   model.Name,
+		Labels: labels,
+	}
+	result, _ := json.Marshal(metadata)
```

Comment:
specifically, I am not sure what I would do with an error there. I would not want to panic the server. Perhaps write an error to the logs and have an additional error return on the function? Suggestions?